tests/test_odcs_requester.py

```diff
@@ -0,0 +1,83 @@
+"""test_odcs_requester.py - test odcs_requester"""
+from typing import Callable
+
+# pylint: disable=redefined-outer-name
```

Comment:
What is this one addressing?
If I recall correctly, if we have disable comments on a line of their own, they will apply to everything below that point in the code. We should instead have them inline (I think), which will only make them apply to the specific line in which they are introduced.

Reply:
It's intended for each time a fixture functions is being passed as a parameter
for instance:
```
def test_odcs_requester(
    create_odcs_mock: Callable,
    compose_url: str,
   ```
   
   I placed it at the top of the file as, same with docstring, I think we should cancel it for tests.

Reply:
Do we need it to be a fixure?

Reply:
I'm not sure I'm following. what do you want to be a function, and why not use a fixture.

Reply:
What do we gain by making this mock-generating function a fixture. The reason not to make it a fixture is that it's (needlessly?) redefines an outer name. Also, if you don't make it a fixture, you could just use it without having it as an input to the test function. I think the test code could stay and we just drop the fixture name from the test inputs.

Am I missing something?

Reply:
We don't really redefine anything, This is the usual way fixtures are to be used... this error will repeat every time we use a fixture in a test function, same error repeats for the other fixtures [1]. I'm pretty sure it was suppressed in `product-ci-cd`, which is why we are not used to see it. 
Why not having it as a fixture? we might want to use it in other tests (for instance, the negative one Gal mentioned).
You cannot drop the fixture name from the test inputs, fixtures need to be mentioned in the test's inputs and not called directly; otherwise you'l get this error [2], which is an issue. 

[1]
```
tests/test_odcs_requester.py:23:21: W0621: Redefining name 'compose_url' from outer scope (line 17) (redefined-outer-name)
tests/test_odcs_requester.py:72:4: W0621: Redefining name 'create_odcs_mock' from outer scope (line 23) (redefined-outer-name)
tests/test_odcs_requester.py:73:4: W0621: Redefining name 'compose_url' from outer scope (line 17) (redefined-outer-name)
tests/test_odcs_requester.py:74:4: W0621: Redefining name 'compose_source' from outer scope (line 48) (redefined-outer-name)
```

[2]
```
Failed: Fixture "create_odcs_mock" called directly. Fixtures are not meant to be called directly
```

Reply:
This is not the first fixture to be used on this repo, and we're not getting this error for any other fixture, so it's not an issue for just any fixture.

Reply:
Test class solved it for this one. not sure it'll act the same way when not using test classes.